src/LatencyEstimator.hpp

```diff
@@ -82,6 +86,8 @@ class LatencyEstimator : public IEstimator {
     Timestamp output_ts_ {};
     Timestamp input_ts_ {};
     SmaCounter sma_;
+
+    const std::unique_ptr<IFormatter>& format_;
```

Comment:
Storing a reference or pointer to `unique_ptr<IFormatter>` doesn't provide any extra safety compared to storing raw reference or pointer to `IFormatter`. If we don't store the unique_ptr by value, we don't get the guarantees its intended to provide (retain the ownership).

Such usage works in this case but is confusing: why we use unique_ptr here at all? is it a bug? It's better either to store unique_ptr by value (if we want to take the ownership) or to store a raw reference / pointer to IFormatter (if we don't need the ownership).

Also, sorry, I spoke too soon and my suggestion to use constant reference was actually meaningless. I was thinking about shared_ptr, in this case passing by constant reference would be okay, it'd allow us to make a copy of the shared_ptr and store it in a field. However unique_ptr can't be copied, it can be only moved.

Passing unique_ptr somewhere makes sense only when we want to *transfer* ownership. This can be done either by passing it by value and using std::move on the caller side, or by passing it by lvalue (`&`) or rvalue (`&&`) reference and using std::move in the called function.

The last two options are usually *not* preferred to transfer ownership because they don't make the fact of the transferring obvious. When you read the call, you see that a unique_ptr is passed by reference and thus *may* be modified in the called function, but you don't know whether the it will be actually moved or not, without reading the source code or documentation of the function.

So the good options here actually are:

* store raw reference, pass raw reference (ownership not transferred)
* store unique_ptr by value, pass unique_ptr by value, use std::move outside of ctor (ownership is transferred)